onadata/libs/filters.py

```diff
@@ -330,18 +337,16 @@ def _xform_filter(self, request, view, keyword):
                 "Invalid value for dataview ID. It must be a positive integer."
             )
             self.dataview = get_object_or_404(DataView, pk=dataview)
-            kwargs = get_filter_kwargs(self.dataview.query)
-            return {
-                **kwargs,
-                **{f"{keyword}": self.dataview.xform}
-            }
+            # filter with fitlered dataset query
+            dataview_kwargs = self._add_instance_prefix_to_dataview_filter_kwargs(
+                get_filter_kwargs(self.dataview.query))
+            xform_qs = XForm.objects.filter(pk=self.dataview.xform.pk)
         if merged_xform:
```

Comment:
Lets use if/elif. I don't think we expect to have a dataview, merged_xform and xform right ?

Reply:
Also the returned queryset for `xform_qs` at the moment will always end up as `XForm.objects.all()` due to line 358 if `xform` is non-truthy